errgo_handler.go

```diff
@@ -0,0 +1,52 @@
+package microerror
+
+import (
+	"fmt"
+
+	"github.com/juju/errgo"
+)
+
+// ErrgoHandler implements Handler interface.
+type ErrgoHandler struct {
+	callDepth int
+	maskFunc  func(err error, allow ...func(error) bool) error
+}
+
+func NewErrgoHandler() *ErrgoHandler {
+	return NewErrgoHandlerCallDepth(0)
+}
+
+// NewErrgoHandlerCallDepth is useful when creating a wrapper for ErrgoHandler.
+// The callDepth parameter is used to push stack location and skip wrapping
+// function location as an origin. The default value is 0.
+func NewErrgoHandlerCallDepth(callDepth int) *ErrgoHandler {
```

Comment:
AFAIK it is common to have `NewDefaultErrgoHandler`, which takes no arguments. Then you have `NewErrgoHandler(Config)`, which takes a config object and the default config object is provided by `DefaultErrgoConfig`. I would go with something like this. Having methods that take one very specific aspect of configuration feels wrong to me, even though it is unlikely to have more options to configure. I like the consistency in our code where we have the `DefaultConfig` and `New` methods. You can then also only do `New(DefaultConfig())`, but the design and the extendability and thus maintainability is good.